loadtest/setup_server.go

```diff
@@ -70,14 +70,40 @@ func SetupServer(cfg *LoadTestConfig) (*ServerSetupData, error) {
 			mlog.Error("Bulk-loading supports at most one plugin deployment at this time")
 		}
 
+		pluginId := "com.mattermost.sample-plugin"
 		plugin, err := os.Open("testfiles/com.mattermost.sample-plugin-webapp-only.tar.gz")
 		if err != nil {
 			return nil, err
 		}
 
+		mlog.Info(fmt.Sprintf("Checking if plugin is installed: %s", pluginId))
+		remotePlugins, resp := adminClient.GetPlugins()
+		if resp.Error != nil {
+			mlog.Error("Failed to get list of installed plugins", mlog.Err(resp.Error))
+			return nil, resp.Error
+		}
+		if len(remotePlugins.Active) > 0 {
+			mlog.Info(fmt.Sprintf("Plugin found, removing: %s", pluginId))
+			for _, remotePlugin := range append(remotePlugins.Active, remotePlugins.Inactive...) {
+				if remotePlugin.Id == pluginId {
+					status, resp := adminClient.RemovePlugin(pluginId)
```

Comment:
Is it worth just trying `RemovePlugin` unconditionally, and checking for an error id of `app.plugin.not_installed.app_error`? Normally, I'd say just test for `http.StatusNotFound`, but it looks like the server code incorrectly returns `http.StatusBadRequest` in this case. (If you want to help patch /that/, feel free!)

Reply:
@irbrad - thoughts?

Reply:
That means the API docs are a little out of sync as well then.  For this PR I'd suggest removing `return nil, resp.Error` on line 92 and leaving the error message line above it.  I'll update this PR and check if there is an issue around for the API and documentation update.

Reply:
I opened mattermost/mattermost-server#11627 to track the API and documentation change and left it unassigned.  It should be a simple fix but I'm going to leave it unassigned until I get a couple minutes to get into that project.  :)

Good catch!

Reply:
👍